app/controllers/expenses_controller.rb

```diff
@@ -1,5 +1,6 @@
 class ExpensesController < ApplicationController
   authorize_resource
+  before_action :setup, only: [:new, :create]
```

Comment:
I would not put this in a `before_action`, because I feel it "hides" behavior, that is important to understand the action.